pulp_container/app/registry.py

```diff
@@ -318,7 +486,54 @@ async def get_by_digest(self, request):
                 "Docker-Content-Digest": ca_content.digest,
             }
         except ObjectDoesNotExist:
-            raise PathNotResolved(path)
+            distribution = await distribution.acast()
```

Comment:
you need to do something about this code being repeated 3 times in 3 places

Reply:
I extracted a new class.